src/neo/IO/Json/JObject.cs

```diff
@@ -56,142 +45,154 @@ public bool ContainsProperty(string key)
+            {
+                AllowTrailingCommas = false,
+                CommentHandling = JsonCommentHandling.Skip,
+                MaxDepth = max_nest
+            });
+            try
             {
-                JObject json = Parse(reader, max_nest);
-                SkipSpace(reader);
-                if (reader.Read() != -1) throw new FormatException();
+                JObject json = Read(ref reader);
+                if (reader.Read()) throw new FormatException();
                 return json;
             }
+            catch (JsonException ex)
+            {
+                throw new FormatException(ex.Message, ex);
+            }
         }
 
-        private static JObject ParseNull(TextReader reader)
+        public static JObject Parse(string value, int max_nest = 100)
         {
-            for (int i = 0; i < LITERAL_NULL.Length; i++)
-                if ((char)reader.Read() != LITERAL_NULL[i])
-                    throw new FormatException();
-            return null;
+            return Parse(Encoding.UTF8.GetBytes(value), max_nest);
         }
 
-        private static JObject ParseObject(TextReader reader, int max_nest)
+        private static JObject Read(ref Utf8JsonReader reader, bool skipReading = false)
         {
-            SkipSpace(reader);
-            if (reader.Read() != BEGIN_OBJECT) throw new FormatException();
-            JObject obj = new JObject();
-            SkipSpace(reader);
-            if (reader.Peek() != END_OBJECT)
+            if (!skipReading && !reader.Read()) throw new FormatException();
+            return reader.TokenType switch
             {
-                while (true)
-                {
-                    string name = JString.Parse(reader).Value;
-                    if (obj.Properties.ContainsKey(name)) throw new FormatException();
-                    SkipSpace(reader);
-                    if (reader.Read() != NAME_SEPARATOR) throw new FormatException();
-                    JObject value = Parse(reader, max_nest - 1);
-                    obj.Properties.Add(name, value);
-                    SkipSpace(reader);
-                    char nextchar = (char)reader.Read();
-                    if (nextchar == VALUE_SEPARATOR) continue;
-                    if (nextchar == END_OBJECT) break;
-                    throw new FormatException();
-                }
-            }
-            else
-            {
-                reader.Read();
-            }
-            return obj;
+                JsonTokenType.False => false,
+                JsonTokenType.Null => Null,
+                JsonTokenType.Number => reader.GetDouble(),
+                JsonTokenType.StartArray => ReadArray(ref reader),
+                JsonTokenType.StartObject => ReadObject(ref reader),
+                JsonTokenType.String => reader.GetString(),
+                JsonTokenType.True => true,
+                _ => throw new FormatException(),
```

Comment:
This could throw `InvalidOperationException`

https://github.com/neo-project/neo/pull/1315/files#diff-f3a55906a56cc27c615edc3f16982666R67

Reply:
Fixed.